packages/augur-ui/src/modules/market/components/common/market-title.tsx

```diff
@@ -45,18 +54,86 @@ const MarketTemplateTitle: React.FC<MarketTemplateTitleProps> = ({
 }) => {
   if (!template || !template.inputs) return null;
   const convertedInputs: TemplateInput[] = template.inputs.map(i => ({
-    userInput: (i.type === TemplateInputType.ESTDATETIME || i.type === TemplateInputType.DATETIME)
-      ? convertUnixToFormattedDate(Number(i.timestamp)).formattedLocalShortDateTimeNoTimezone
-      : i.value,
+    userInput:
+      i.type === TemplateInputType.ESTDATETIME ||
+      i.type === TemplateInputType.DATETIME
+        ? convertUnixToFormattedDate(Number(i.timestamp))
+            .formattedLocalShortDateTimeNoTimezone
+        : i.value,
     id: i.id,
     type: i.type as TemplateInputType,
     placeholder: '',
   }));
-  const question = buildMarketDescription(template.question, convertedInputs);
+  let question = buildMarketDescription(template.question, convertedInputs);
   const estDateTime = convertedInputs.find(
     i => i.type === TemplateInputType.ESTDATETIME
   );
 
+  const hasInputs = hasTemplateTextInputs(template.hash);
+  if (hasInputs) {
+    const originalQuestion = template.question.split(' ');
+    let prevWordUnique = false;
+    question = originalQuestion.map((word, index) => {
```

Comment:
Looks a bit over complicated. See if you can simplify it. Will be hard to maintain if more changes are needed